core/src/main/java/no/nav/foreldrepenger/vtp/felles/AzureOidcTokenGenerator.java

```diff
@@ -0,0 +1,108 @@
+package no.nav.foreldrepenger.vtp.felles;
+
+import com.google.common.base.Strings;
+import org.jose4j.jwk.RsaJsonWebKey;
+import org.jose4j.jws.AlgorithmIdentifiers;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.jwt.JwtClaims;
+import org.jose4j.jwt.NumericDate;
+import org.jose4j.lang.JoseException;
+
+import java.util.*;
+
+
+public class AzureOidcTokenGenerator {
+
+
+    private List<String> aud = Arrays.asList(
+            "OIDC"
+    );
+    private NumericDate expiration = NumericDate.fromSeconds(NumericDate.now().getValue() + 3600*6);
+    private String issuer;
+    private NumericDate issuedAt = NumericDate.now();
+    private final String subject;
+    private String kid = KeyStoreTool.getJsonWebKey().getKeyId();
+    private String nonce;
+    private Map<String, String> additionalClaims = new HashMap<>();
+
+    public AzureOidcTokenGenerator(String brukerId, String nonce) {
+        additionalClaims.put("azp", "OIDC");
+        additionalClaims.put("acr", "Level4");
+        this.subject = brukerId;
+        this.nonce = nonce;
+
+    }
+
+    public void addAud(String e) {
+        this.aud = new ArrayList<>(aud);
+        this.aud.add(e);
+    }
+
+    AzureOidcTokenGenerator withoutAzp() {
+        additionalClaims.remove("azp");
+        return this;
+    }
+
+    AzureOidcTokenGenerator withExpiration(NumericDate expiration) {
+        this.expiration = expiration;
+        return this;
+    }
+
+    public AzureOidcTokenGenerator withIssuer(String issuer) {
+        this.issuer = issuer;
+        return this;
+    }
+
+    AzureOidcTokenGenerator withIssuedAt(NumericDate issuedAt) {
+        this.issuedAt = issuedAt;
+        return this;
+    }
+
+    AzureOidcTokenGenerator withKid(String kid) {
+        this.kid = kid;
+        return this;
+
+    }
+
+    AzureOidcTokenGenerator withClaim(String name, String value) {
+        additionalClaims.put(name, value);
+        return this;
+    }
+
+    AzureOidcTokenGenerator withAud(List<String> aud) {
+        this.aud = aud;
+        return this;
+    }
+
+    public String create() {
+        JwtClaims claims = new JwtClaims();
+        claims.setIssuer(issuer);
+        claims.setExpirationTime(expiration);
+        claims.setGeneratedJwtId();
+        claims.setIssuedAt(issuedAt);
+        claims.setSubject(subject);
+        if (!Strings.isNullOrEmpty(nonce)) {
```

Comment:
Trengs guava for dette?  Kunne man brukt (Objects.isNull(nonce) || nonce.isBlank()) som alternativ ?